### PR TITLE
Helper lines object counter

### PR DESCRIPTION
**Pull request by Dgzt**
Show the selected helper line object (rectangle or hexagon) on the status bar.

counter

How this works in Tiled:
tiled

If the helper line is disabled (what is the default value) the helper line object counter will not be show on the status bar.

I added 2 new field where can enable helper lines.
image

These offset spinners will add offset to the counter.

The helper lines settings will be hide if the helper lines checkbox is disabled:
Peek 2023-08-18 13-26

### CODE REVIEW

File: editor/src/main/com/mbrlabs/mundus/editor/input/FreeCamController.kt [hunk 1]
```diff
@@ -166,4 +171,33 @@ class FreeCamController : InputAdapter() {
...
+
+        val currentProject = projectManager.current()
+        val currentScene = currentProject.currScene
+
+        val ray = currentScene.viewport.getPickRay(screenX.toFloat(), screenY.toFloat())
+
+        var helperCell: HelperLineCenterObject? = null
+
+        for (terrain in currentScene.terrains) {
+            val result = getRayIntersection(terrain, ray, Vector3())
```
Comment (JamesTKhan) [inline editor/src/main/com/mbrlabs/mundus/editor/input/FreeCamController.kt:188]: Do we need to do all this calculation logic on mouse move when helper lines are disabled?

Comment (Dgzt) [reply editor/src/main/com/mbrlabs/mundus/editor/input/FreeCamController.kt]: Thank you for your review. I rewrite that codes to use GameObjectPicker and added getRayIntersection method to TerrainComponent what calls Terrain's getRayIntersection method.

**Comment by JamesTKhan** [comment]
Looks good, I am hesitant as this is very project specific features but I see no harm in adding it. I may rework the debug rendering later to break the debug render options into separate windows.

**Comment by Dgzt** [comment]
> Looks good, I am hesitant as this is very project specific features but I see no harm in adding it. I may rework the debug rendering later to break the debug render options into separate windows.

I tried to made it general as I could that anyone else could use it, but later if you think we can remove the helper lines feature or part of this feature, because I can use it in my fork.

**Pull request merged**